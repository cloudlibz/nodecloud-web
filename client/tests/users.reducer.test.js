import * as reducer from "../src/_reducers/users.reducer";

describe("reducer", () => {
  describe("GETALL REQUEST", () => {
    test("sets getall to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_GETALL_REQUEST"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("GETALL SUCCESS", () => {
    test("sets getall to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_GETALL_SUCCESS"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("GETALL FAILURE", () => {
    test("sets getall to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_GETALL_FAILURE"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("DELETE REQUEST", () => {
    test("sets delete to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_DELETE_REQUEST"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("DELETE SUCCESS", () => {
    test("sets delete to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_DELETE_SUCCESS"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("DELETE FAILURE", () => {
    test("sets delete to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_DELETE_FAILURE"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE VIRTUAL MACHINE REQUEST", () => {
    test("sets create virtual machine to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_VIRTUAL_MACHINE_REQUEST"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE VIRTUAL MACHINE SUCCESS", () => {
    test("sets create virtual machine to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_VIRTUAL_MACHINE_SUCCESS"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE VIRTUAL MACHINE FAILURE", () => {
    test("sets create virtual machine to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_VIRTUAL_MACHINE_FAILURE"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE VIRTUAL NETWORK REQUEST", () => {
    test("sets create virtual network to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_VIRTUAL_NETWORK_REQUEST"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE VIRTUAL NETWORK SUCCESS", () => {
    test("sets create virtual network to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_VIRTUAL_NETWORK_SUCCESS"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE VIRTUAL NETWORK FAILURE", () => {
    test("sets create virtual network to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_VIRTUAL_NETWORK_FAILURE"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE DATBASE REQUEST", () => {
    test("sets create database to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_DATBASE_REQUEST"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE DATBASE SUCCESS", () => {
    test("sets create database to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_DATBASE_SUCCESS"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE DATBASE FAILURE", () => {
    test("sets create database to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_DATBASE_FAILURE"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE SECURITY REQUEST", () => {
    test("sets create security to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_SECURITY_REQUEST"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE SECURITY SUCCESS", () => {
    test("sets create security to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_SECURITY_SUCCESS"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("CREATE SECURITY FAILURE", () => {
    test("sets create security to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_CREATE_SECURITY_FAILURE"
      };
      const nextState = reducer.users(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });
});
