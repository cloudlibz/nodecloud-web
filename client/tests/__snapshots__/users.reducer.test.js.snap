// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer CREATE DATBASE FAILURE sets create database to failure 1`] = `
Object {
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE DATBASE REQUEST sets create database to requesting 1`] = `
Object {
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE DATBASE SUCCESS sets create database to success 1`] = `
Object {
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE SECURITY FAILURE sets create security to failure 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE SECURITY REQUEST sets create security to requesting 1`] = `
Object {
  "loading": true,
}
`;

exports[`reducer CREATE SECURITY SUCCESS sets create security to success 1`] = `
Object {
  "items": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE VIRTUAL MACHINE FAILURE sets create virtual machine to failure 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE VIRTUAL MACHINE REQUEST sets create virtual machine to requesting 1`] = `
Object {
  "loading": true,
}
`;

exports[`reducer CREATE VIRTUAL MACHINE SUCCESS sets create virtual machine to success 1`] = `
Object {
  "items": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE VIRTUAL NETWORK FAILURE sets create virtual network to failure 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer CREATE VIRTUAL NETWORK REQUEST sets create virtual network to requesting 1`] = `
Object {
  "loading": true,
}
`;

exports[`reducer CREATE VIRTUAL NETWORK SUCCESS sets create virtual network to success 1`] = `
Object {
  "items": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer DELETE FAILURE sets delete to failure 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer DELETE REQUEST sets delete to requesting 1`] = `
Object {
  "loading": true,
}
`;

exports[`reducer DELETE SUCCESS sets delete to success 1`] = `
Object {
  "items": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;

exports[`reducer GETALL FAILURE sets getall to failure 1`] = `
Object {
  "error": undefined,
  "loading": false,
}
`;

exports[`reducer GETALL REQUEST sets getall to requesting 1`] = `
Object {
  "loading": true,
}
`;

exports[`reducer GETALL SUCCESS sets getall to success 1`] = `
Object {
  "items": undefined,
  "loading": false,
  "otherProp": "test-123",
}
`;
